dat.js

```diff
@@ -0,0 +1,117 @@
+var assert = require('assert')
+var path = require('path')
+var multicb = require('multicb')
+var importFiles = require('./lib/import-files')
+var createNetwork = require('./lib/network')
+var stats = require('./lib/stats')
+
+module.exports = Dat
+
+function Dat (archive, db, opts) {
+  if (!(this instanceof Dat)) return new Dat(archive, db, opts)
+  if (typeof opts === 'undefined') return Dat(archive, null, db)
+  assert.ok(archive, 'archive required')
+  // assert.ok(db, 'database required') // maybe not be required for multidrive...
+  assert.ok(opts.dir, 'opts.directory required')
+
+  this.path = path.resolve(opts.dir)
+  this.options = opts
+
+  this.archive = archive
+  this.db = db
+  this.key = archive.key // only resumed/owned archives will have keys here
+  this.live = archive.live
+  this.owner = archive.owner
```

Comment:
Do I remember it right that `archive.owner` is only populated after `archive.open`?